src/components/settings/SettingsForm.tsx

```diff
@@ -0,0 +1,316 @@
+                Setup your Plainly API key to communicate directly with Plainly
+                Videos. Your API key can be found in the{' '}
+                <button
+                  type="button"
+                  className="underline text-white"
+                  onClick={handleLinkClick.bind(
+                    null,
+                    'https://app.plainlyvideos.com/dashboard/user/settings/general',
+                  )}
+                >
+                  Settings page
+                </button>{' '}
+                at Plainly.
+              </Description>
+            </div>
+            <div className="mt-2 grid grid-cols-1">
+              <input
+                id="api-key"
+                name="api-key"
+                type={showApiKey ? 'text' : 'password'}
+                className="col-start-1 row-start-1 block w-full rounded-md bg-white/5 pl-3 pr-16 py-1 text-xs text-white outline outline-1 -outline-offset-1 outline-white/10 placeholder:text-gray-500 focus:outline focus:outline-2 focus:-outline-offset-2 focus:outline-indigo-500"
+                value={apiKey}
+                onChange={(e) => setApiKey(e.target.value)}
+                required
+              />
+              <button
+                type="button"
+                onClick={() => {
+                  navigator.clipboard.writeText(apiKey || '');
+                  notifySuccess('API key copied to clipboard');
+                }}
+                className="col-start-1 row-start-1 mr-9 self-center justify-self-end cursor-pointer w-4 h-4 flex items-center justify-center group"
+              >
+                <CopyIcon className="size-4 shrink-0 text-gray-400 group-hover:text-white" />
+              </button>
+              <button
+                type="button"
+                onClick={() => setShowApiKey((prev) => !prev)}
+                className="col-start-1 row-start-1 mr-3 self-center justify-self-end cursor-pointer w-4 h-4 flex items-center justify-center group"
+              >
+                {showApiKey ? (
+                  <EyeIcon className="size-4 shrink-0 text-gray-400 group-hover:text-white" />
+                ) : (
+                  <EyeOffIcon className="size-4 shrink-0 text-gray-400 group-hover:text-white" />
+                )}
+              </button>
+            </div>
+          </div>
+
+          <div className="col-span-full">
+            <div>
+              <Label label="PIN (recommended)" htmlFor="pin" />
+              <Description>
+                Setup your four digit PIN to encrypt your API key stored on your
+                device. You will be asked to enter this PIN on first request
+                sent to Plainly every time you restart the extension. If you
+                don&apos;t want to use a PIN, you can leave this field empty.
+              </Description>
+            </div>
+            <div className="mt-2 grid grid-cols-6 items-center justify-items-center gap-y-2">
+              <PinInput
```

Comment:
* please show me the pin here always
* and return that confirmation text, as I know don't know what it means

![image](https://github.com/user-attachments/assets/bcb37445-1761-4e61-9ea2-de53800a4062)